language-support/ts/codegen/src/TsCodeGenMain.hs

```diff
@@ -528,92 +570,74 @@ writePackageIdTs dir pkgId =
       ["export default '" <> unPackageId pkgId <> "';"]
 
 writeTsConfig :: FilePath -> IO ()
-writeTsConfig dir = writeFileUTF8 (dir </> "tsconfig.json") $ unlines
-    [ "{"
-    , "  \"compilerOptions\": {"
-    , "    \"target\": \"es5\","
-    , "    \"lib\": ["
-    , "      \"dom\","
-    , "      \"es2015\""
-    , "     ],"
-    , "    \"strict\": true,"
-    , "    \"noUnusedLocals\": true,"
-    , "    \"noUnusedParameters\": false,"
-    , "    \"noImplicitReturns\": true,"
-    , "    \"noFallthroughCasesInSwitch\": true,"
-    , "    \"outDir\": \"lib\","
-    , "    \"module\": \"commonjs\","
-    , "    \"declaration\": true,"
-    , "    \"sourceMap\": true"
-    , "  },"
-    , "  \"include\": [\"src/**/*.ts\"],"
-    , "}"
-    ]
+writeTsConfig dir =
+  BSL.writeFile (dir </> "tsconfig.json") $ encodePretty tsConfig
+  where
+    tsConfig :: Value
+    tsConfig = object
+      [ "compilerOptions" .= object
+        [ "target" .= ("es5" :: T.Text)
+        , "lib" .= (["dom", "es2015"] :: [T.Text])
+        , "strict" .= True
+        , "noUnusedLocals" .= True
```

Comment:
Phantom type parameters will produce unused locals. That's why it was `false` before.
```suggestion
        , "noUnusedLocals" .= False
```

Reply:
genuine bug.